src/System.ServiceModel.Primitives/tests/ServiceModel/DuplexChannelFactoryTest.cs

```diff
@@ -0,0 +1,384 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Reflection;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using System.ServiceModel.Tests.Common;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+public class DuplexChannelFactoryTest
+{
+    [Fact]
+    public static void CreateChannel_EndpointAddress_Null_Throws()
+    {
+        WcfDuplexServiceCallback callback = new WcfDuplexServiceCallback();
+        Binding binding = new NetTcpBinding();
+        EndpointAddress remoteAddress = null;
+
+        DuplexChannelFactory<IWcfDuplexService> factory = new DuplexChannelFactory<IWcfDuplexService>(callback, binding, remoteAddress);
+        try {
+            Assert.Throws<InvalidOperationException>(() =>
+               {
+                   Assert.Equal(CommunicationState.Created, factory.State);
+                   factory.Open();
+                   factory.CreateChannel();
```

Comment:
Not a problem but this test reminds me we need to test async opens and async closes.